core/src/main/java/io/substrait/extendedexpression/ExtendedExpressionProtoConverter.java

```diff
@@ -0,0 +1,64 @@
+package io.substrait.extendedexpression;
+
+import io.substrait.expression.Expression;
+import io.substrait.expression.proto.ExpressionProtoConverter;
+import io.substrait.extension.ExtensionCollector;
+import io.substrait.proto.ExpressionReference;
+import io.substrait.proto.ExtendedExpression;
+import io.substrait.relation.AggregateFunctionProtoConverter;
+import io.substrait.type.proto.TypeProtoConverter;
+
+/**
+ * Converts from {@link io.substrait.extendedexpression.ExtendedExpression} to {@link
+ * io.substrait.proto.ExtendedExpression}
+ */
+public class ExtendedExpressionProtoConverter {
+  public ExtendedExpression toProto(
+      io.substrait.extendedexpression.ExtendedExpression extendedExpression) {
+
+    ExtendedExpression.Builder builder = ExtendedExpression.newBuilder();
+    ExtensionCollector functionCollector = new ExtensionCollector();
+
+    final ExpressionProtoConverter expressionProtoConverter =
+        new ExpressionProtoConverter(functionCollector, null);
+
+    for (io.substrait.extendedexpression.ExtendedExpression.ExpressionReferenceBase
+        expressionReference : extendedExpression.getReferredExpressions()) {
+      if (expressionReference
+          instanceof io.substrait.extendedexpression.ExtendedExpression.ExpressionReference et) {
+        io.substrait.proto.Expression expressionProto =
+            expressionProtoConverter.visit(
+                (Expression.ScalarFunctionInvocation) et.getExpression());
```

Comment:
Why are we restricting this to scalar functions invocations? Are the other expression types not valid here?

Reply:
Ah, I see, I'm only considering Filter and Projection cases. Let me review changes needed to be more generic.

Reply:
Changed